pyresample/boundary/boundary_sides.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2023 Pyresample developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Define the BoundarySides class."""
+
+import logging
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+class BoundarySides:
+    """A class to represent the sides of an area boundary.
+
+    The sides are stored as a tuple of 4 numpy arrays, each representing the
+    coordinate (geographic or projected) of the vertices of the boundary side.
+    The sides must be stored in the order (top, right, left, bottom),
+    which refers to the side position with respect to the coordinate array.
+    The first row of the coordinate array correspond to the top side, the last row to the bottom side,
+    the first column to the left side and the last column to the right side.
+    Please note that the last vertex of each side must be equal to the first vertex of the next side.
+    """
+    __slots__ = ['_sides']
+
+    def __init__(self, sides: tuple[ArrayLike, ArrayLike, ArrayLike, ArrayLike]):
+        """Initialize the BoundarySides object."""
+        if len(sides) != 4 or not all(isinstance(side, np.ndarray) and side.ndim == 1 for side in sides):
+            raise ValueError("Sides must be a list of four numpy arrays.")
```

Comment:
Instead of controlling the size of the argument, why not pass four arguments instead? this would make it also clear which one is top, bottom, etc...

Reply:
I defined like this because in `geometry.py` we always retrieve directly a list with the 4 elements. 
Do you think it make sense to unpack? 
Also consider that for some projections (the left and right) sides are just "dummy" and of size 2 (because we have a circle ...) ...

Reply:
This is such an obvious solution it makes me angry that I didn't think of it in the giant pull request. I think the fact that left and right are sometimes "dummy" sides doesn't change anything unless there is some optimization by knowing that information, but that doesn't seem to be used here, right? ...I still think a namedtuple serves the same purpose.

I haven't used it yet, but I'm 90% sure that you can specify shape or dimensionality and dtype with the `ArrayLike` type annotation. That might add some more type-annotation-level information for the user and IDEs to make sure the right thing is passed.